metadata_service/api/user.py

```diff
@@ -31,8 +36,211 @@ def __init__(self) -> None:
+                               'is not added successfully'.format(user_id,
+                                                                  table_uri)}, \
+                HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+class UserOwnAPI(Resource):
+    """
+    Build get / put API to support user own resource features.
+    It will create a relationship(owner / owner_of) between user and resources(table, dashboard etc)
+    todo: Deprecate TableOwner API
+    """
+
+    def __init__(self) -> None:
+        self.neo4j = neo4j_proxy.get_neo4j()
+
+    def get(self, user_id: str) -> Tuple:
+        """
+        Return a list of resources that user has owned
+
+        :param user_id:
+        :return:
+        """
+        try:
+            resources = self.neo4j.get_resources_by_user_relation(user_id=user_id,
+                                                                  relation='OWNER_OF')
+            return marshal(resources, table_list_fields), HTTPStatus.OK
+
+        except NotFoundException:
+            return {'message': 'user_id {} does not exist'.format(user_id)}, HTTPStatus.NOT_FOUND
+
+        except Exception:
+            return {'message': 'Internal server error!'}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+    def put(self, user_id: str, table_uri: str) -> Iterable[Union[Mapping, int, None]]:
+        try:
+            self.neo4j.add_owner(table_uri=table_uri,
+                                 owner=user_id)
+            return {'message': 'The owner {} for table_uri {} '
+                               'is added successfully'.format(user_id,
+                                                              table_uri)}, HTTPStatus.OK
+        except Exception as e:
+            return {'message': 'The owner {} for table_uri {} '
+                               'is not added successfully'.format(user_id,
+                                                                  table_uri)}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+    def delete(self, user_id: str, table_uri: str) -> Iterable[Union[Mapping, int, None]]:
+        try:
+            self.neo4j.delete_owner(table_uri=table_uri,
+                                    owner=user_id)
+            return {'message': 'The owner {} for table_uri {} '
+                               'is deleted successfully'.format(user_id,
+                                                                table_uri)}, HTTPStatus.OK
+        except Exception:
+            return {'message': 'The owner {} for table_uri {} '
+                               'is not deleted successfully'.format(user_id,
+                                                                    table_uri)}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+class UserReadAPI(Resource):
+    """
+    Build get / put API to support user bookmark resource features.
+    It will create a relationship(read / read_by) between user and resources(table, dashboard etc)
+
+    todo: do we need put / delete to this API? And do we need to put a threshold on the get?
```

Comment:
I don't think we have a use case to mutate read record. We can remove it unless we have a use case.

For read, did we define how many tables we will display as well as sorting rule? We may need pagination or truncation

cc: @danwom

Reply:
correct, make sense, let me remove the two mutate methods.

In terms of sorting, let's keep it simple in first pr(I put this qq in this pr description )